src/cryptsetup/cryptsetup.c

```diff
@@ -907,6 +907,7 @@ static int run(int argc, char *argv[]) {
                         return log_error_errno(r, "crypt_init() failed: %m");
 
                 crypt_set_log_callback(cd, cryptsetup_log_glue, NULL);
+                crypt_set_debug_level(DEBUG_LOGGING ? CRYPT_DEBUG_ALL : CRYPT_DEBUG_NONE);
```

Comment:
maybe we should actually take these two lines and make them a helper in src/shared/crypt-util.h: cryptsetup_log_setup() or so, that does both in one?

Reply:
Sure, done